inc/Engine/Optimization/DeferJS/DeferJS.php

```diff
@@ -75,6 +75,83 @@ public function defer_js( string $html ) : string {
 		return $html;
 	}
 
+	/**
+	 * Defers inline JS containing jQuery calls
+	 *
+	 * @since 3.8
+	 *
+	 * @param string $html HTML content.
+	 * @return string
+	 */
+	public function defer_inline_js( string $html ) : string {
+		if ( ! $this->can_defer_js() ) {
+			return $html;
+		}
+
+		$exclude_defer_js = implode( '|', $this->get_excluded() );
+
+		if ( preg_match( '/(jquery(?:.*)?\.js)/i', $exclude_defer_js ) ) {
+			return $html;
+		}
+
+		$buffer_nocomments = preg_replace( '/<!--(.*)-->/Uis', '', $html );
+
+		preg_match_all( '#<script(?:[^>]*)>(?<content>[\s\S]*?)</script>#msi', $buffer_nocomments, $matches, PREG_SET_ORDER );
+
+		if ( empty( $matches ) ) {
+			return $html;
+		}
+
+		/**
+		 * Filters the patterns used to find jQuery in inline JS
+		 *
+		 * @since 3.8
+		 *
+		 * @param string $jquery_patterns A RegEx pattern to find jQuery inline JS.
+		 */
+		$jquery_patterns = apply_filters( 'rocket_defer_jquery_patterns', 'jQuery|\$\.\(|\$\(' );
+
+		/**
+		 * Filters the patterns used to find inline JS that should not be deferred
+		 *
+		 * @since 3.8
+		 *
+		 * @param string $inline_exclusions A RegEx pattern to find inline JS that should not be deferred.
+		 */
+		$inline_exclusions = apply_filters( 'rocket_defer_inline_exclusions', 'DOMContentLoaded|document\.write' );
```

Comment:
I don't know if this is valid case or not but the following two cases will fail:-
```
<script type="text/javascript">
	jQuery('document').ready(function(){
		//document.write('Hi');
	});
</script>
<script type="text/javascript">
	document.writename = function(){
	    alert('asa');
	};
	jQuery('document').ready(function(){
	    document.writename();
	});
</script>
```
Both of these cases will not be enclosed inside `DOMContentLoaded` event and I think they should.

Reply:
Do you see a way of making the cases pass while complying with the technical specs?

Reply:
for the commented code the solution will be expensive but for `writename` I'd adjust the regex to use write and writeln only.
after thinking about that, if it made a problem support can guide client how to use the filter.